### PR TITLE
Block Library: Unify handling for block view scripts

### PR DESCRIPTION
**Pull request by gziolo**
### Compatibility between the plugin and WordPress core

The following code won't work in WordPress core, because `plugins_urls` can only work with the Gutenberg plugin. We need a more general solution for core blocks to move forward:

```js
wp_enqueue_script( 'wp-block-library-file', plugins_url( 'file/frontend.js', __FILE__ ) );
```

My proposal is that we only call `wp_enqueue_script` with the script handle and we register script separately outside of the PHP file that gets copied to WordPress core. This part is still missing in this PR.

### Rename `frontend` to `view`

Based on the feedback from @mtias, I included the change that promotes the name `view` for the script to make it more general. The reasoning is that the output generated by blocks can be consumed not only the frontend part of WordPress but you could render them inside the WP-Admin.

## How has this been tested?

I tested with the Navigation block and its "Enable responsive menu" option enabled in the sidebar menu.

Screen Shot 2021-06-21 at 12 45 44

Once you insert the block, enable the option. It has to be tested on the front end on small viewports.

It should be tested also with the File block and embed previews for PDF files.

## Screenshots 

Screen Shot 2021-06-21 at 12 42 55
Screen Shot 2021-06-21 at 12 43 09

Screen Shot 2021-06-21 at 12 43 15

## Types of changes

Bug fix (non-breaking change which fixes an issue).

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: lib/blocks.php [hunk 1]
```diff
@@ -180,7 +206,7 @@ function gutenberg_register_core_block_styles( $block_name ) {
 			"wp-block-{$block_name}",
 			gutenberg_url( $style_path ),
 			array(),
-			filemtime( gutenberg_dir_path() . $style_path )
+			$default_version
```
Comment (gziolo) [inline lib/blocks.php:209]: @aristath – I copied the logic from other CSS files used in Gutenberg that ensures that the plugin version is used on production.

**Comment by gziolo** [comment]
@vcanales and @jasmussen, I would appreciate a sanity check from you for the Navigation block and in regards to all the changes applied to how this mechanism works now to account for the fact that the same PHP files power both the block in the Gutenberg plugin and in WordPress core.

@pento, I would appreciate a sanity check for the File block. As part of this PR I also sorted out the issue you raised in the past about the potential name conflict between packages and blocks by using the following name patterns for entry points:

```js
[ 'blocks/' + blockName ]: scriptPath,
```

**Review by jorgefilipecosta — APPROVED** [review]
Things worked well on my tests and the change seems a good improvement 👍 
I tested that the navigation block continues to be responsive. For the file block, I added some alerts and verified they were shown in the browser (so the script loaded).

**Comment by jorgefilipecosta** [comment]
This should be backported to core right?

**Comment by youknowriad** [comment]
What about the e2e test failures here, are they unrelated? (the navigation one specifically)

File: lib/blocks.php (hunk 1 above)
Comment (aristath) [reply lib/blocks.php:209]: Makes sense :+1:

**Review by aristath — APPROVED** [review]

**Review by pento — APPROVED** [review]
Changes for the file block look solid, thanks for handling that, @gziolo!

**Comment by mtias** [comment]
Thanks for looking at this. As we introduce core blocks that have a reliance on frontend scripts it's important to ensure the APIs created move in the right direction.

**Comment by gziolo** [comment]
> What about the e2e test failures here, are they unrelated? (the navigation one specifically)

I'm looking into it now. It feels like an issue with e2e tests related to changes. It's most likely my fault 😄

**Comment by gziolo** [comment]
It looks like failing PHPUnit tests are related to the changes to the template editing mode in WP core, related PR: #32888.

**Comment by gziolo** [comment]
> It looks like failing PHPUnit tests are related to the changes to the template editing mode in WP core, related PR: #32888.

Resolved after rebase with `trunk` that contains a fix to PHPUnit tests.

It looks like failing e2e tests are completely unrelated to this branch after the fix added in 89ba0a6 for the tests covering Navigation block.

**Review by vcanales — APPROVED** [review]
Tested and it works as expected on the Navigation and File blocks. e2e test failures seem unrelated as well, as they're [present in trunk](https://github.com/WordPress/gutenberg/runs/2884909993) at the moment.

Thanks for looking into this, this process is a great improvement.

**Pull request merged**